packages/salesforcedx-vscode-core/test/vscode-integration/commands/functions/forceFunctionInvokeCodeLensProvider.test.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { expect } from 'chai';
+import { CancellationTokenSource, RelativePattern, workspace } from 'vscode';
+import { provideFunctionInvokeCodeLens as unit } from '../../../../src/commands/functions';
+import { getRootWorkspacePath } from '../../../../src/util';
+
+describe('Force Apex Function Invoke Codelens', () => {
+
+  it('should create codelens for a json payload file', async () => {
+    const cancellationTokenSource = new CancellationTokenSource();
+    workspace.openTextDocument({
```

Comment:
I suppose this creates an untitled document. Would codelens show up in untitled JSON? (You can create an untitled doc and go to command palette -> change language mode)

Reply:
This created an untitled.json doc and hence gets picked up to show the codelens. I tried to change the language mode of a different file type from command palate -> change language mode to JSON, but codelens don't appear in that case.

Reply:
I was wanting to confirm here that if you create an untitled doc (cmd + N) in VS Code and change language mode to JSON it should not show any CodeLens, since document selector should have limited it to functions folder. If that doesn't show up it's fine

Reply:
Otherwise, you might want to add `scheme: 'file'` to document selector

Reply:
Ah! That's a such an edge case to take care :)
Thankfully it does not show the codelens. That's because the filename is still ~Untitled even when we change the language mode.

Reply:
Sounds good, thanks for checking! I was wondering how this test was able to generate codelens

Reply:
Good question. For some reason, the behavior is different here. When you set language as json, it changes the filename to Untitled.json.